server/api.go

```diff
@@ -853,3 +847,163 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request, userID s
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request, userID string) {
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	githubClient := p.githubConnect(*info.Token)
+
+	ctx := context.Background()
+	org := p.getConfiguration().GitHubOrg
+
+	var allRepos []*github.Repository
+	opt := github.ListOptions{PerPage: 50}
+
+	if org == "" {
+		for {
+			repos, resp, err := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{ListOptions: opt})
+			if err != nil {
+				mlog.Error(err.Error())
+				return
+			}
+			allRepos = append(allRepos, repos...)
+			if resp.NextPage == 0 {
+				break
+			}
+			opt.Page = resp.NextPage
+		}
+	} else {
+		for {
+			repos, resp, err := githubClient.Repositories.ListByOrg(ctx, org, &github.RepositoryListByOrgOptions{Sort: "full_name", ListOptions: opt})
+			if err != nil {
+				mlog.Error(err.Error())
+				return
+			}
+			allRepos = append(allRepos, repos...)
+			if resp.NextPage == 0 {
+				break
+			}
+			opt.Page = resp.NextPage
+		}
+	}
+
+	resp, _ := json.Marshal(allRepos)
+	w.Write(resp)
+}
+
+func (p *Plugin) createIssue(w http.ResponseWriter, r *http.Request, userID string) {
+	type IssueRequest struct {
+		Title  string `json:"title"`
+		Body   string `json:"body"`
+		Repo   string `json:"repo"`
+		PostId string `json:"post_id"`
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		mlog.Error("Error decoding JSON body", mlog.Err(err))
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Title == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue title.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Repo == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid repo name.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.PostId == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a postID", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	ghIssue := &github.IssueRequest{
+		Title: &issue.Title,
+		Body:  &issue.Body,
+	}
+
+	permalink := p.getPermaLink(issue.PostId)
+
+	mmMessage := fmt.Sprintf("_Issue created from a [message in Mattermost](%v)_.", permalink)
```

Comment:
@jfrerich Mentioning the author of the message would help make the context of the message more clear. But I'm failing a bit on the wording.  

My initial suggestion was `Issue created from a message in Mattermost from @hanzei`, but it feels weird to have `from` two times in the sentence.

Reply:
Hmmm.. I thought about several other options, but believe yours is quite simple and correct.  How about the following?

`Issue created from a [Mattermost message] by @hanzei`